plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/TestMongoConnectorTest.java

```diff
@@ -341,7 +341,11 @@ public Object[][] predicatePushdownProvider()
                 {"smallint '2'"},
                 {"integer '3'"},
                 {"bigint '4'"},
+                {"DECIMAL '3.14'"},
```

Comment:
Apart from this - can we update the tests in `TestMongoTypeMapping` ?

Reply:
As discussed offline, we wanted to fix below two cases. But Decimal128 supports 34 digits precision. and below tests still failing with the current change.

```
// TODO: Fix NumberFormatException: Conversion to Decimal128 would require inexact rounding
//                .addRoundTrip("decimal(38, 0)", "CAST('27182818284590452353602874713526624977' AS decimal(38, 0))", createDecimalType(38, 0), "CAST('27182818284590452353602874713526624977' AS decimal(38, 0))")
//                .addRoundTrip("decimal(38, 0)", "CAST('-27182818284590452353602874713526624977' AS decimal(38, 0))", createDecimalType(38, 0), "CAST('-27182818284590452353602874713526624977' AS decimal(38, 0))")
```

Reply:
As discussed offline. To ensure 38 precision on the filter should not error,  I have added `testHighPrecisionDecimalPredicate`